### PR TITLE
Fix flakes in tests related to Cloud Foundry

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

Fixes flakes in tests related to Cloud Foundry functionality.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by gzussa — DISMISSED** [review]

**Review by olivielpeau — DISMISSED** [review]
a couple of notes/suggestions, no need to address them now

File: pkg/autodiscovery/listeners/cloudfoundry_test.go [hunk 1]
```diff
@@ -388,7 +388,7 @@ func TestCloudFoundryListener(t *testing.T) {
 		testBBSCache.DesiredLRPs = tc.dLRP
 		testBBSCache.Unlock()
 
-		time.Sleep(20 * time.Millisecond)
+		time.Sleep(50 * time.Millisecond)
 		// we have to fail now, otherwise we might get blocked trying to read from the channel
 		if !assert.Equal(t, len(tc.expNew), len(newSvc)) {
```
[the remark this answers is not in the document]
Comment (bkabrda) [reply pkg/autodiscovery/listeners/cloudfoundry_test.go]: That's awesome, I didn't know about this. I'll amend this PR to get this improved. Thanks!

File: pkg/autodiscovery/listeners/cloudfoundry_test.go [hunk 2]
```diff
@@ -388,7 +388,7 @@ func TestCloudFoundryListener(t *testing.T) {
 		testBBSCache.DesiredLRPs = tc.dLRP
 		testBBSCache.Unlock()
 
-		time.Sleep(20 * time.Millisecond)
+		time.Sleep(50 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (bkabrda) [reply pkg/autodiscovery/listeners/cloudfoundry_test.go]: Yeah, agreed. I think it shouldn't flake any more, as I found out a specific case with the sleep time <=20 ms and I think anything above 20 ms should work. If it doesn't I'll use the retrier.

**Review by gzussa — APPROVED** [review]
LGTM

**Comment by bkabrda** [comment]
Thanks for the reviews! Merging.

**Pull request merged**